### PR TITLE
react_add-todo-form

### PR DESCRIPTION
**Pull request by svyatoslavPy**
[DEMO LINK](https://svyatoslavPy.github.io/react_add-todo-form/)

### CODE REVIEW

**Review by anastasiiavorobiova — CHANGES_REQUESTED** [review]
Make sure to deploy your solution

File: src/App.tsx [hunk 1]
```diff
@@ -1,61 +1,105 @@
...
+  };
+
+  const handlerSumbit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!nameUser) {
+      setErrorNameUser(true);
+    }
+
+    if (!title) {
```
Comment (anastasiiavorobiova) [inline src/App.tsx:46]:
```suggestion
    if (!title.trim()) {
```

File: src/App.tsx [hunk 2]
```diff
@@ -1,61 +1,105 @@
...
+      setErrorTitle(true);
+    }
+
+    if (!nameUser || !title) {
```
Comment (anastasiiavorobiova) [inline src/App.tsx:50]:
```suggestion
    if (!nameUser || !title.trim()) {
```

File: src/App.tsx [hunk 3]
```diff
@@ -1,61 +1,105 @@
...
+import todosFromServer from './api/todos';
+
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+import { TodoList } from './components/TodoList';
 
 export const App = () => {
+  const [todos, setTodos] = useState<Todo[]>(todosFromServer);
+  const [title, setTitle] = useState<string>('');
+  const [nameUser, setNameUser] = useState<string>('');
```
Comment (anastasiiavorobiova) [inline src/App.tsx:14]: Consider using userId here

**Review by anastasiiavorobiova — CHANGES_REQUESTED** [review]
Great start! Keep doing! Consider finding users for todos from the server

File: src/App.tsx [hunk 4]
```diff
@@ -1,61 +1,105 @@
...
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+import { TodoList } from './components/TodoList';
 
 export const App = () => {
+  const [todos, setTodos] = useState<Todo[]>(todosFromServer);
```
Comment (anastasiiavorobiova) [inline src/App.tsx:12]: Consider normalizing data from the server and attaching the user for the todo

File: src/App.tsx (hunk 3 above)
Comment (anastasiiavorobiova) [inline src/App.tsx:14]: Consider using state for userIds instead of names

**Review by anastasiiavorobiova — CHANGES_REQUESTED** [review]
Great work! Some comments are still not fixed. Consider also clearing the form after the submit 
Screenshot 2024-04-16 at 17 15 08

File: src/components/TodoInfo/TodoInfo.tsx [hunk 5]
```diff
@@ -1 +1,19 @@
...
+
+export const TodoInfo: React.FC<TodoInfoProps> = ({ todo }) => {
+  return (
+    <article
+      data-id={todo.id}
+      className={cn('TodoInfo', {
+        'TodoInfo--completed': todo.completed,
+      })}
+    >
+      <h2 className="TodoInfo__title">{todo.title}</h2>
```
Comment (volodymyr-soltys97) [inline src/components/TodoInfo/TodoInfo.tsx:14]: Use destruction for `todo`

File: src/types/Todo.ts [hunk 6]
```diff
@@ -0,0 +1,9 @@
+import { User } from './User';
+
+export type Todo = {
+  id: number;
+  title: string;
+  completed: boolean;
+  userId?: number;
+  user?: User | null;
```
Comment (volodymyr-soltys97) [inline src/types/Todo.ts:8]:
```suggestion
  user: User;
```

File: src/components/UserInfo/UserInfo.tsx [hunk 7]
```diff
@@ -1 +1,9 @@
-export const UserInfo = () => {};
+import { UserInfoProps } from '../../types/UserInfo';
+
+export const UserInfo: React.FC<UserInfoProps> = ({ user }) => {
+  return (
+    <a className="UserInfo" href={`mailto:${user?.email}`}>
+      {user?.name}
```
Comment (volodymyr-soltys97) [inline src/components/UserInfo/UserInfo.tsx:6]: Also, use a destruction for `user`
```suggestion
    <a className="UserInfo" href={`mailto:${user.email}`}>
      {user.name}
```

**Review by volodymyr-soltys97 — CHANGES_REQUESTED** [review]
Almost done!
Let's make your code better
1. You need to add label for inputs as on the working example, pls check it
image
image

**Review by l4st1m0za — APPROVED** [review]
Good job!

**Pull request closed without merging**